File: common/src/main/java/dev/obscuria/fragmentum/api/v1/common/FragmentumEvents.java
package dev.obscuria.fragmentum.api.v1.common;

import dev.obscuria.fragmentum.api.v1.common.event.Event;
import net.minecraft.world.inventory.tooltip.TooltipComponent;
import net.minecraft.world.item.ItemStack;

import java.util.function.Consumer;

@SuppressWarnings("unused")
public interface FragmentumEvents
{
    Event<GatherTooltipImages> GATHER_TOOLTIP_IMAGES = Event.create();

    @FunctionalInterface
    interface GatherTooltipImages
    {
        void invoke(ItemStack stack, Consumer<TooltipComponent> consumer);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/FragmentumEvents.kt
package dev.obscuria.fragmentum.api

import dev.obscuria.fragmentum.api.tools.event.Event
import net.minecraft.world.inventory.tooltip.TooltipComponent
import net.minecraft.world.item.ItemStack
import java.util.function.Consumer

object FragmentumEvents
{
    val GATHER_TOOLTIP_IMAGES: Event<GatherTooltipImages> = Event.create()

    fun interface GatherTooltipImages
    {
        fun invoke(stack: ItemStack, consumer: Consumer<TooltipComponent>)
    }
}